torch/_subclasses/functional_tensor.py

```diff
@@ -308,10 +308,17 @@ def unwrap(x):
         )
         assert is_excluded or not is_included
 
-        # If no outputs are our functional subclass, then don't try to fix up aliasing
-        if not any(
-            isinstance(x, FunctionalTensor)
-            for x in pytree.tree_flatten(outs_wrapped)[0]
+        if (
+            # If no outputs are our functional subclass, then don't try to fix up aliasing
+            not any(
+                isinstance(x, FunctionalTensor)
+                for x in pytree.tree_flatten(outs_wrapped)[0]
+            )
+            # Since lift_fresh lifts its argument into a functional tensor, we can skip the
+            # aliasing correction step. Otherwise, we would be setting the storage of a
+            # lifted tensor to that of an unlifted tensor.
+            # Ref: https://github.com/pytorch/pytorch/issues/111506
+            or func == torch.ops.aten.lift_fresh.default
```

Comment:
Hmm, I think that if the func here is `lift_fresh`, you might need to manually wrap the output in a `FunctionalTensor()`.

Reply:
But, we are already returning the wrapped version of the outputs, right!?

https://github.com/pytorch/pytorch/blob/21330e5ba19b4f22afbd26168eb72bcd8586071d/torch/_subclasses/functional_tensor.py#L298

Reply:
oh yep you're right!